src/validation.cpp

```diff
@@ -1045,6 +1051,11 @@ bool CChainState::IsInitialBlockDownload() const
 
 static CBlockIndex *pindexBestForkTip = nullptr, *pindexBestForkBase = nullptr;
 
+BlockMap& CChainState::BlockIndex() const
```

Comment:
In commit "refactoring: remove mapBlockIndex global" (1e33f16dd47c42a9f6099f6172ebd5f8fbb32ec8)

I'm probably just bikeshedding, but I think having this method is counterproductive and only obfuscates what's going on. In places where this is called in `CChainState` methods it seems clearer to replace `BlockIndex()` calls with direct `m_blockman.m_block_index` references. In places where this is called globally, it would seem clearer to able to just write `::BlockIndex()` instead of `::ChainstateActive().BlockIndex()` since the block index doesn't depend on the active chain, and because it would be more consistent with the existing `::LookupBlockIndex` calls.

Reply:
Yep, I agree with this. Willfix.